frontend/taipy-gui/src/components/Taipy/FileSelector.tsx

```diff
@@ -68,14 +68,14 @@ const FileSelector = (props: FileSelectorProps) => {
         defaultLabel = "",
         updateVarName = "",
         multiple = false,
-        selectFolder = false,
+        selectionType = "file",
         extensions = ".csv,.xlsx",
         dropMessage = "Drop here to Upload",
         label,
         notify = true,
         withBorder = true,
     } = props;
-    const directoryProps = selectFolder ? {webkitdirectory: "", directory: "", mozdirectory: "", nwdirectory: ""}:undefined;
+    const directoryProps = ["d", "dir", "directory", "folder"].includes(selectionType) ? {webkitdirectory: "", directory: "", mozdirectory: "", nwdirectory: ""} : undefined;
```

Comment:
I would check for selectionType?.toLowerCase().startsWith("d") || selectionType?.toLowerCase()).startsWith("folder")